File: backend/app/schemas/logs.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class OperationLogResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    level: str
    module: str
    message: str
    detail: str | None
    created_at: datetime


class OperationLogListResponse(BaseModel):
    message: str
    logs: list[OperationLogResponse]

File: frontend/src/types/logs.ts
export type OperationLogLevel = 'info' | 'warning' | 'error'

export interface OperationLog {
  id: number
  level: OperationLogLevel
  module: string
  message: string
  detail: string | null
  created_at: string
}

export interface OperationLogListResponse {
  message: string
  logs: OperationLog[]
}
